pipelines/extension-build.yml

```diff
@@ -18,36 +15,83 @@ pr:
     - /extensions/
     - /pipelines/extension-build.yml
 
-steps:
-- task: NPM@1
-  displayName: npm install
-  inputs:
-    command: install
-    workingDir: extensions
-
-- task: NPM@1
-  displayName: npm run lint
-  inputs:
-    command: custom
-    customCommand: run lint
-    workingDir: extensions
-
-- task: NPM@1
-  displayName: npm run build
-  inputs:
-    command: custom
-    customCommand: run build
-    workingDir: extensions
-
-- task: CopyFiles@2
-  displayName: Stage extension artifact for publish
-  inputs:
-    sourceFolder: extensions
-    contents: '**/*.vsix'
-    targetFolder: $(Build.ArtifactStagingDirectory)
-
-- task: PublishPipelineArtifact@0
-  displayName: Publish extension artifact
-  inputs:
-    artifactName: drop
-    targetPath: $(Build.ArtifactStagingDirectory)
+stages:
+- stage: Build
+  jobs:
+  - job: buildvsix
+    pool:
+      vmImage: 'ubuntu-latest' 
+    steps:
+    - task: NPM@1
+      displayName: npm install
+      inputs:
+        command: install
+        workingDir: extensions
+
+    - task: NPM@1
+      displayName: npm run lint
+      inputs:
+        command: custom
+        customCommand: run lint
+        workingDir: extensions
+
+    - task: NPM@1
+      displayName: npm run build
+      inputs:
+        command: custom
+        customCommand: run build
+        workingDir: extensions
+
+    - task: CopyFiles@2
+      displayName: Stage extension artifact for publish
+      inputs:
+        sourceFolder: extensions
+        contents: '**/*.vsix'
+        targetFolder: $(Build.ArtifactStagingDirectory)
+
+    - task: PublishPipelineArtifact@0
+      displayName: Publish extension artifact
+      inputs:
+        artifactName: drop
+        targetPath: $(Build.ArtifactStagingDirectory)
+
+    - task: ms-devlabs.vsts-developer-tools-build-tasks.tfx-installer-build-task.TfxInstaller@2
+      displayName: 'Use Node CLI for Azure DevOps (tfx-cli): v0.6.x'
+
+    - task: ms-devlabs.vsts-developer-tools-build-tasks.publish-extension-build-task.PublishAzureDevOpsExtension@2
+      displayName: 'Publish Extension'
+      inputs:
+        connectedServiceName: marketplaceConnection
+        fileType: vsix
+        vsixFile: '**/*.vsix'
+        publisherId: NLUDevOps
+        extensionId: NLUDevOpsCI
+        extensionName: NLU.DevOps.CI
+        extensionVersion: '0.1.$(Build.BuildId)'
+        updateTasksVersion: true
+        extensionVisibility: private
+        shareWith: NLUDevOps
+
+- stage: Wait
+  dependsOn: Build
+  jobs:
+  - job: waitforvsix
+    pool: server
+    steps:
+    - task: Delay@1
+      inputs: 
+        delayForMinutes: 1
+
+- stage: Trigger
+  dependsOn: Wait
+  jobs:
+  - job: triggertest  
+    steps: 
+    - task: maikvandergaag.maikvandergaag-trigger-pipeline.TriggerPipeline.TriggerPipeline@1
```

Comment:
Should we just use a build completion trigger in the visual/classic pipeline editor as a stop-gap until there is YAML for triggering a pipeline on build completion, rather than taking a dependency on the TriggerPipeline task?

Reply:
Added build completion trigger